### PR TITLE
login-logout

### PR DESCRIPTION
**Pull request by hatcaiday**
https://edu-redmine.framgia.vn/issues/5015

### CODE REVIEW

**Comment by hatcaiday** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+module SessionsHelper
+  def log_in user
+  	session[:user_id] = user.id
```
Comment (huongnguyenmta) [inline app/helpers/sessions_helper.rb]: indent

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  def remember_user user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent.remember_token = user.remember_token
+  end
+
+  def current_user? user
```
Comment (huongnguyenmta) [inline app/helpers/sessions_helper.rb]: định nghĩa method này trong model User em nhé

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      user = User.find_by id: user_id
+      if user&.authenticated? :remember, cookies[:remember_token]
+        log_in user
+        @current_user = user
+      end
+    end
+  end
+
+  def logged_in?
+  	current_user.present?
```
Comment (huongnguyenmta) [inline app/helpers/sessions_helper.rb]: indent

File: app/helpers/sessions_helper.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  def forget user
+    user.forget
+    cookies.delete :user_id
+    cookies.delete :remember_token
+  end
+
+  def log_out
+  	forget current_user
```
Comment (huongnguyenmta) [inline app/helpers/sessions_helper.rb]: indent

**Comment by huongnguyenmta** [comment]
commented

**Comment by hatcaiday** [comment]
ready

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -53,18 +53,22 @@
...
-                <% end %>
-              </li>
               <li>
                 <% if logged_in? %>
-                  <%= link_to root_path do %>
-                    <i class="fa fa-lock"></i><%= t ".logout" %>
-                  <% end %>
+                  <li class="dropdown">
+                    <%= link_to "#", class: "dropdown-toggle", data: {toggle: "dropdown"}  do %>
+                      <i class="fa fa-user"></i><%= get_name %>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: No need to write a method to get the logged in user name. 
Just check if the user is logged_in? and than write
```ruby
<%= current_user.username %>
```

**Comment by mdmahmudurrahman** [comment]
(commented)

File: app/views/layouts/_header.html.erb (hunk 5 above)
Comment (hatcaiday) [reply app/views/layouts/_header.html.erb]: ohhh. thank U :)

**Comment by hatcaiday** [comment]
ready

**Comment by mdmahmudurrahman** [comment]
(merged)

**Pull request merged**